custom/icds_reports/filters.py

```diff
@@ -37,38 +37,63 @@ def loc_to_json(loc, project):
 
     project = Domain.get_by_name(domain)
 
-    locations = SQLLocation.root_locations(domain)
+    if user.has_permission(domain, 'access_all_locations'):
+        accessible_root_locations = SQLLocation.root_locations(domain)
+    else:
+        accessible_root_locations = [(user.sql_location
+                                      .get_ancestors(include_self=True)
+                                      .get(location_type__code='state'))
+                                     ]
```

Comment:
nit: if the sql_location of the user is a root location on its own, this basically does not getting anything new i.e the case where the sql_location is itself the top most location, is implicitly handled here. One can skip getting the ancestors if the sql_location is itself the root, but might not be worth. (just noted this down as a observation, there are pros and cons to both implementations)

```
user_primary_location = user.sql_location
if user_primary_location.parent_id is None:
    accessible_root_locations = [user_primary_location]
else:
    accessible_root_locations = [(user.sql_location.get_ancestors().filter(parent_id__isnull=True)]
```

But worth considering `parent_id__isnull=True`, in case `state` is not the topmost location anymore, sometime in future though that will break many other things.